test/InterversionTests/RavenDB-20628-backward-compatibility.cs

```diff
@@ -0,0 +1,158 @@
+            var followerUrl = nodes.Single(n => n.Url != leaderUrl).Url;
+
+            using (result.Disposable)
+            {
+                var stores = result.Stores;
+                await CreateDatabase(stores, database, nodes.Count);
+                await UpgradeServerAsync("current", leaderProc);
+
+                using var followerStore = new DocumentStore()
+                {
+                    Urls = new[] { followerUrl },
+                    Conventions =
+                    {
+                        DisableTopologyUpdates = true
+                    }
+                }.Initialize();
+
+                var user1 = new User() { Id = "Users/1-A", Name = "Alice" };
+
+                using (var session = followerStore.OpenAsyncSession(new SessionOptions
+                       {
+                           Database = database,
+                           TransactionMode = TransactionMode.ClusterWide
+                       }))
+                {
+                    await session.StoreAsync(user1);
+                    await session.SaveChangesAsync();
+                }
+            }
+        }
+
+        [MultiplatformFact(RavenPlatform.Windows | RavenPlatform.Linux)]
+        public async Task UpgradeFollowerToLatest()
+        {
+            DebuggerAttachedTimeout.DisableLongTimespan = true;
+
+            var latest = "5.4.107";
+            var initialVersions = new[] { latest, latest };
+
+            var nodes = await CreateCluster(initialVersions, watcherCluster: true);
+            var database = GetDatabaseName();
+            var result = await GetStores(database, nodes);
+            var leaderUrl = await GetLeaderUrl(result.Stores[0]);
+            var followerProc = nodes.Single(n => n.Url != leaderUrl);
+            var followerUrl = followerProc.Url;
+
+            using (result.Disposable)
+            {
+                var stores = result.Stores;
+                await CreateDatabase(stores, database, nodes.Count);
+                await UpgradeServerAsync("current", followerProc);
+
+                using var followerStore = new DocumentStore()
+                {
+                    Urls = new[] { followerUrl },
+                    Conventions =
+                    {
+                        DisableTopologyUpdates = true
+                    }
+                }.Initialize();
+
+                var user1 = new User() { Id = "Users/1-A", Name = "Alice" };
+
+                using (var session = followerStore.OpenAsyncSession(new SessionOptions
+                       {
+                           Database = database,
+                           TransactionMode = TransactionMode.ClusterWide
+                       }))
+                {
+                    await session.StoreAsync(user1);
+                    await session.SaveChangesAsync();
+                }
+            }
+        }
+
+        private static async Task CreateDatabase(List<DocumentStore> stores, string database, int size)
```

Comment:
don't we have a method to create a database?

Reply:
Done